lib/overcommit/hook/pre_commit/line_endings.rb

```diff
@@ -0,0 +1,42 @@
+module Overcommit::Hook::PreCommit
+  # Checks for line endings in files.
+  class LineEndings < Base
+    def run
+      messages = []
+
+      text_files.map do |file_name|
+        file = File.open(file_name)
+        file.each_line do |line|
+          next if unix? && line =~ /\A((?!\r).)*\n\z/
+          next if windows? && line =~ /\A.*\r\n\z/
```

Comment:
Perhaps there's a nuance here I'm missing, but is there any reason we can't use `String#end_with?` here? For example:

```ruby
next if unix? && !line.end_with?("\r\n")
next if windows? && line.end_with?("\r\n")
```

Performing a regex match for every line in a file will be much slower than running `end_with?`. For example:

```ruby
require 'benchmark'

iterations = 1_000_000
line = "this is a line\r\n"

Benchmark.bmbm do |x|
  x.report('regex') do
    iterations.times { line =~ /\A((?!\r).)*\n\z/ }
  end

  x.report('end_with?') do
    iterations.times { line.end_with?("\r\n") }
  end
end
```
```
Rehearsal ---------------------------------------------
regex       0.910000   0.000000   0.910000 (  0.912726)
end_with?   0.120000   0.000000   0.120000 (  0.120560)
------------------------------------ total: 1.030000sec

                user     system      total        real
regex       0.870000   0.000000   0.870000 (  0.875747)
end_with?   0.130000   0.000000   0.130000 (  0.126313)
```

These optimizations might not seem like much, but when you're running `overcommit --run` on a repository with many large files the performance will matter.